include/seastar/kafka/producer/kafka_producer.hh

```diff
@@ -41,7 +41,7 @@ private:
 
     std::string _client_id;
     lw_shared_ptr<connection_manager> _connection_manager;
-    partitioner* _partitioner = new basic_partitioner();
+    std::unique_ptr<partitioner> _partitioner;
```

Comment:
Is there a reason for this to be allocated dynamically?

Reply:
It has to be this way if we want to use ABCs, because they cannot be instantiated.